tests/test_format.py

```diff
@@ -430,3 +468,52 @@ def test_xml_key_error():
 
     with pytest.raises(FormatError, match="key not found 'foo'"):
         format.get_values(file, [Key("foo")])
+
+
+@pytest.mark.h5
+def test_bzip2_h5py():
+    h5_buffer = io.BytesIO()
+    with h5py.File(h5_buffer, "w") as f:
+        f["foo"] = "foo value"
+        f["bar"] = "bar value"
+        f["list"] = ["list", "value"]
+
+    bz2_compressed_bytes = bz2.compress(h5_buffer.getvalue())
+    bz2_compressed_file = io.BytesIO(bz2_compressed_bytes)
```

Comment:
I think, just use `file` like the rest of the tests do.

```suggestion
    file = io.BytesIO(bz2.compress(h5_buffer.getvalue()))
```

Reply:
I think I prefer when reading the test its clearer to name the variable `bz2_compressed_file` I could make it into a single call though